amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisher.java

```diff
@@ -489,7 +490,13 @@ private void callShutdownOnStrategy() {
         }
 
         private void sleepBeforeNextCall() throws InterruptedException {
-            if (lastSuccessfulCall == null) {
+            if (lastSuccessfulCall == null && isFirstGetCallTry) {
+                isFirstGetCallTry = false;
+                return;
+            }
+            // Add a sleep if lastSuccessfulCall is still null but this is not the first try to avoid retry storm
+            if(lastSuccessfulCall == null) {
+                Thread.sleep(idleMillisBetweenCalls);
                 return;
             }
             long timeSinceLastCall = Duration.between(lastSuccessfulCall, Instant.now()).abs().toMillis();
```

Comment:
Is lastSuccessfulCall set correctly when we're getting throttled? If we make a successful call and only receiving throttling events onwards, I think we can get stuck in a throttling loop here.